be/test/exprs/vectorized/decimal_binary_function_test.cpp

```diff
@@ -2837,4 +2955,121 @@ TEST_F(DecimalBinaryFunctionTest, test_decimal128p38s14_div_decimal128p38s14_eq_
+                                       {"533.04", "3067942701.0884", "1635336177388.160736"},
+                                       {"2742930.04", "-99999834076370.4796", "-274292548883092242664.047184"},
+                                       {"2269330.25", "0.0361", "81922.822025"}};
+    test_decimal_fast_mul_help<TYPE_DECIMAL32, TYPE_DECIMAL64, TYPE_DECIMAL128, MulOp32x64_128>(test_cases, 9, 2, 18, 4,
+                                                                                                38, 6);
+    test_decimal_fast_mul_help<TYPE_DECIMAL64, TYPE_DECIMAL64, TYPE_DECIMAL128, MulOp64x64_128>(test_cases, 9, 2, 18, 4,
+                                                                                                38, 6);
+}
+
+TEST_F(DecimalBinaryFunctionTest, test_decimal_fast_mul_64x64) {
+    DecimalTestCaseArray test_cases = {
+            {"0.0006", "-82780021658097.3483", "-49668012994.85840898"},
+            {"0.0074", "-99827652105771.3123", "-738724625582.70771102"},
+            {"-99973172481064.8851", "41055058.0121", "-4104404395863798146564.72090971"},
+            {"10418.0297", "-78182709607827.6816", "-814509790720824139.39094352"},
+            {"84231788195839.5714", "77738262261.3072", "6548032841507056635908796.85173408"},
+            {"-95593465131926.9776", "-2201368772639.3279", "210436469009810477989501604.74235504"},
+            {"-5829918858943.4969", "41160.0999", "-239960042643008340.85934031"},
+            {"22286515224419.0423", "-45199673982539.5695", "-1007343222350645401189049609.04428985"},
+            {"0.3026", "-67285817488933.7073", "-20360688372151.33982898"},
+            {"93643928070236.3651", "-85952437612924.2777", "-8048923885286159719807127647.19098827"},
+            {"-68455776580233.7557", "5711626623.7667", "-390993836066288058523463.22159519"},
+            {"96190766579211.5819", "-99999999999998.3166", "-9619076657920996262463540555.22302954"},
+            {"-99999999911670.1121", "-7286176685749.2421", "728617667931337040128175917.79703941"},
+            {"5300.1993", "-14319371956134.8862", "-75895525218345754.54281966"},
+            {"52771758641568.0008", "-25416153656949.7962", "-1341255126381560552930609374.00143696"},
+            {"3.2917", "-99999999999999.9213", "-329169999999999.74094321"},
+            {"-78809072751576.0845", "-14758895507471.3475", "1163134869781218860114475849.98886375"},
+            {"-63650347027092.4971", "83952744775493.0151", "-5343621338837057001558324362.66700621"},
+            {"197070.9764", "20234892726480.4061", "3987710066956751765.59551604"},
+            {"0.0114", "-75288876496539.0058", "-858293192060.54466612"},
+            {"-72638466358742.9662", "3.1225", "-226813611205174.91195950"},
+            {"60915079068521.5269", "-99999121308263.0016", "-6091454381275516575941632856.40914304"}};
+    test_decimal_fast_mul_help<TYPE_DECIMAL64, TYPE_DECIMAL64, TYPE_DECIMAL128, MulOp64x64_128>(test_cases, 18, 4, 18,
```

Comment:
Are there some overflow cases?

Reply:
this situation never overflows, try to think, i64 x i64 yields i128, can it overflow?